src/app/pages/send-tokens/hooks/use-send-form-validation.ts

```diff
@@ -124,8 +152,9 @@ export const useSendFormValidation = ({
         memo: transactionMemoSchema(SendFormErrorMessages.MemoExceedsLimit),
         nonce: nonceSchema,
         recipient: recipientSchema(),
+        recipientAddressOrBnsName: recipientAddressOrBnsNameSchema,
```

Comment:
Nice approach using another form field, this could work for now.

Were this implemented greenfield, I wonder if it'd be cleaner to have some custom address vs bns name detection and conditionally routing the validation logic.

Reply:
If I understood correctly, I believe it's what's being done in this PR,

https://github.com/hirosystems/stacks-wallet-web/blob/b36034e9a31ca64492d74f08331e988201c4f91b/src/app/pages/send-tokens/hooks/use-send-form-validation.ts#L107-L131

Code can probably benefit from some cleanup, but the idea here is to first check whether the address is a valid, if not, use the input as a bns query, and extract the address from there. If that fails, then the field as a whole fails validation.